auth.go

```diff
@@ -194,12 +194,6 @@ func filterDefinition(sourceSchema *ast.Schema, visited map[string]bool, types m
 			_ = filterDefinition(sourceSchema, visited, types, sourceSchema.Types[typeName], AllowedFields{AllowAll: true})
 		}
 
-		// unions
```

Comment:
What was the reasoning for this step being here originally?

Reply:
From what I could gather, it was a special case to deal with unions but didn't function correctly because it ignored the union's possible types.